### ISSUE TITLE
Exception thrown: 'System.Runtime.InteropServices.COMException' in WindowsFirewallHelper.dll

### ISSUE THREAD

**Issue by GitUser200607** [falahati/WindowsFirewallHelper#68]
I have some code:
```
		foreach (FirewallWASRuleWin8 rule in FirewallManager.Instance.Rules)
		{
			// Delete old rule by rule Name (we don't have GUID in FW rule)
			if (rule.Grouping == $"_{App.ApplicationName}")
			{
				if (rule is not null)
				{
					FirewallManager.Instance.Rules.Remove(rule);
				}
			}
		}

```
The result of code working is good, but when I run Visual Studio Community 2022, I see the errors in Debug:
```
Exception thrown: 'System.Runtime.InteropServices.COMException' in WindowsFirewallHelper.dll
Exception thrown: 'System.NotSupportedException' in WindowsFirewallHelper.dll
```
I've thought it was the problem with the deleting FW rule, but the errors are shown even before condition. Should I ignore them?

_Windows 10 Pro x64 (19045.5247)_

**Comment by kamronbatman** [comment]
To fix it, @falahati will need to change this line:
https://github.com/falahati/WindowsFirewallHelper/blob/ccc858703a425f49b9bca7f0471bdb91977e3ed9/WindowsFirewallHelper/COMTypeResolver.cs#L29

So that instead of `null`, the machine name is `"localhost"`

**Comment by GitUser200607** [comment]
> So that instead of `null`, the machine name is `"localhost"`

Can we change it in our code without modifying WindowsFirewallHelper? Something like overriting the method? This repository was not updated since Jan 10, 2022...

**Comment by kamronbatman** [comment]
I don't think so?

**Comment by falahati** [comment]
I don’t understand why you’re concerned about an exception that is raised and handled internally within the library. If it’s interfering with your debugging, it’s likely because you have “Just My Code” debugging set to false, which is a choice you made, not something caused by the library. However, if you’re actually encountering the “not supported” exception while using the library, please provide more details, including the specific line where the exception occurs, the full stack trace, and OS details such as your Windows version.

Setting that value to localhost prevents the exception from being thrown. This is simply how the logic is implemented, as feature detection for remote machines is not feasible. `null` here indicates not connecting to a remote machine, and therefore has a meaning.

Of course, you are free to always create an instance of the library with `localhost` set as the machine name.

```c#
var instance = new FirewallWAS(new COMTypeResolver("localhost"))
``` 

You can also inherit and change the logic of `COMTypeResolver` however you want.

**Comment by GitUser200607** [comment]
> I don’t understand why you’re concerned about an exception that is raised and handled internally within the library.

I was afraid that there was an error with the library. And “Just My Code” is enabled. But I see the errors wile running project in Debug mode. 

> if you’re actually encountering the “not supported” exception while using the library, please provide more details, including the specific line where the exception occurs, the full stack trace, and OS details such as your Windows version.

How can I do that?

**Comment by falahati** [comment]
> I was afraid that there was an error with the library. And “Just My Code” is enabled. But I see the errors while running project in Debug mode.

Yes, that is how the library internally works when it tries to detect what features are supported. However, these errors should only be thrown on legacy operating systems. Anything above Windows 8 should not result in any error.

> How can I do that?

Based on your description, this error is not actually blocking your debugging process. So if the library works as expected, just ignore them. But if you want to really know where and why this error is raised, you can disable the "Just My Code" and see the actual exception that contains the stack trace and everything else needed. If it is in fact raised mistakenly I might have to fix it.

**Comment by kamronbatman** [comment]
@falahati:

Windows 11
Target Platform x64 (my use case is not Any CPU and we do not support 32bit).
Just My Code is enabled

Happens with Rider, VS 2022, and when any .NET debugger is attached otherwise.

The workaround you provided is good enough, but a fix or documentation would be nice. Thank you!
